src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let switcher = true;
+
+    while (switcher) {
+      switcher = false;
+
+      for (let i = 0; i < this.length - 1; i++) {
+        const firstValue = this[i];
+        const secondValue = this[i + 1];
+        const compareString = (a, b) => (String(a) > String(b) ? 1 : -1);
+
+        const callback =
+          compareFunction === undefined
+            ? compareString(firstValue, secondValue)
+            : compareFunction(firstValue, secondValue);
```

Comment:
The variable `callback` is misleading because it suggests that a function is being called back, but it actually stores the result of the comparison. Consider renaming it to `comparisonResult` or a similar name that clearly indicates its purpose.